library/common/engine.cc

```diff
@@ -64,7 +64,9 @@ envoy_status_t Engine::run(std::string config, std::string log_level) {
     postinit_callback_handler_ = main_common_->server()->lifecycleNotifier().registerCallback(
         Envoy::Server::ServerLifecycleNotifier::Stage::PostInit, [this]() -> void {
           Server::Instance* server = TS_UNCHECKED_READ(main_common_)->server();
-          http_dispatcher_->ready(server->dispatcher(), server->clusterManager());
+          auto api_listener = server->listenerManager().apiListener()->get().http();
+          ASSERT(api_listener.has_value());
+          http_dispatcher_->ready(server->dispatcher(), api_listener.value());
```

Comment:
Is this additional delay (beyond construction) still necessary? We're no longer waiting on an initial round of DNS resolution and the filter dynamic forwarding filter contains built in buffering logic while we do wait. If we're unsure, maybe insert a TODO to follow up.

Reply:
Still suggest adding a TODO and/or issue for this so we still have visibility to follow up.